appserver/data/data-core/src/main/java/fish/payara/jakarta/data/core/cdi/extension/RepositoryImpl.java

```diff
@@ -248,41 +249,56 @@ public Object processInsertOperation(Object[] args, QueryData dataForQuery) thro
+            throws SystemException, NotSupportedException, HeuristicRollbackException,
+            HeuristicMixedException, RollbackException {
         int returnValue = 0;
         Annotation[][] parameterAnnotations = method.getParameterAnnotations();
-        Class<?>[] types = method.getParameterTypes();
-        if (parameterAnnotations.length == 1 && types.length == 1 && parameterAnnotations[0].length == 1) {
-            Annotation[] annotations = parameterAnnotations[0];
-            for (Annotation annotation : annotations) {
-                if (annotation instanceof By) {
-                    //for now we are processing only By id operation, when custom By operation available we will provide 
-                    // the metadata from the entity class to search specific column value for By
-                    String byValue = ((By) annotation).value();
-                    returnValue = processDeleteByIdOperation(args, declaredEntityClass, getTransactionManager(), getEntityManager(this.applicationName), byValue);
-                }
+
+        if (args == null) { // delete all records
+            startTransactionComponents();
+            startTransactionAndJoin();
+            String deleteAllQuery = "DELETE FROM " + declaredEntityClass.getSimpleName();
+            returnValue = em.createQuery(deleteAllQuery).executeUpdate();
+            endTransaction();
+        } else if (args[0] instanceof List arr) {
+            // existing list handling code
+            startTransactionComponents();
+            startTransactionAndJoin();
+            List<Object> ids = getIds((List<?>) arr);
+            if (!ids.isEmpty()) {
+                String deleteQuery = "DELETE FROM " + declaredEntityClass.getSimpleName() + " e WHERE e.id IN :ids";
+                returnValue = em.createQuery(deleteQuery)
+                        .setParameter("ids", ids)
+                        .executeUpdate();
             }
+            endTransaction();
         } else {
-            startTransactionComponents();
-            if (args == null) { //delete all records
-                startTransactionAndJoin();
-                String deleteAllQuery = "DELETE FROM " + declaredEntityClass.getSimpleName();
-                returnValue = em.createQuery(deleteAllQuery).executeUpdate();
-                endTransaction();
-            } else if (args[0] instanceof List arr) {
-                startTransactionAndJoin();
-                List<Object> ids = getIds((List<?>) arr);
-                if (!ids.isEmpty()) {
-                    String deleteQuery = "DELETE FROM " + declaredEntityClass.getSimpleName() + " e WHERE e.id IN :ids";
-                    returnValue = em.createQuery(deleteQuery)
-                            .setParameter("ids", ids)
-                            .executeUpdate();
+            // Handle @By annotation cases
+            boolean hasByAnnotation = false;
+            for (Annotation[] annotations : parameterAnnotations) {
```

Comment:
probably this part of the code can be changed using lambdas and optional:

```suggestion
            Optional<Annotation> byFound = 
                    Arrays.stream(parameterAnnotations).flatMap(Arrays::stream).filter(a -> a instanceof By).findFirst();
```

Reply:
Done!